x-pack/test_serverless/api_integration/test_suites/common/reporting/download_csv_dashboard.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { X_ELASTIC_INTERNAL_ORIGIN_REQUEST } from '@kbn/core-http-common';
+import expect from '@kbn/expect';
+import { INTERNAL_ROUTES } from '@kbn/reporting-plugin/common/constants';
+import { JobParamsDownloadCSV } from '@kbn/reporting-plugin/server/export_types/csv_searchsource_immediate/types';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+const ELASTIC_USERNAME = 'elastic';
+const ELASTIC_PASSWORD = 'changeme';
+const API_HEADER: [string, string] = ['kbn-xsrf', 'reporting'];
+const INTERNAL_HEADER: [string, string] = [X_ELASTIC_INTERNAL_ORIGIN_REQUEST, 'Kibana'];
+const getMockJobParams = (obj: object) => {
+  return {
+    title: `Mock CSV Title`,
+    ...obj,
+  } as JobParamsDownloadCSV;
+};
+
+export default function ({ getService }: FtrProviderContext) {
+  const kibanaServer = getService('kibanaServer');
+  const supertestSvc = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  const generateAPI = {
+    downloadCsvFromSearchSource: async (job: JobParamsDownloadCSV) => {
+      return await supertestSvc
+        .post(INTERNAL_ROUTES.DOWNLOAD_CSV)
+        .auth(ELASTIC_USERNAME, ELASTIC_PASSWORD)
+        .set(...API_HEADER)
+        .set(...INTERNAL_HEADER)
+        .send(job);
+    },
+  };
+
+  const archives = {
+    data: 'x-pack/test/functional/es_archives/reporting/ecommerce', // ecommerce data
+    savedObjects: 'x-pack/test/functional/fixtures/kbn_archiver/reporting/ecommerce', // ecommerce saved objects (no Canvas)
+  };
+  const fromTime = '2019-06-20T00:00:00.000Z';
+  const toTime = '2019-06-25T00:00:00.000Z';
+
+  describe('CSV Generation from SearchSource: Dashboard', () => {
```

Comment:
Let's skip this until the Core test gets fixed

Reply:
Sounds good to me! Opened https://github.com/elastic/kibana/issues/164867 for tracking https://github.com/elastic/kibana/issues/164867